sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/AesKw.cs

```diff
@@ -0,0 +1,422 @@
+    /// </summary>
+    internal abstract class AesKw
+    {
+        private const int BlockSizeInBits = 64;
+        private const int BlockSizeInBytes = BlockSizeInBits >> 3;
+
+        private static readonly byte[] _defaultIv = new byte[] { 0xA6, 0xA6, 0xA6, 0xA6, 0xA6, 0xA6, 0xA6, 0xA6 };
+
+        protected static RandomNumberGenerator Rng = RandomNumberGenerator.Create();
+
+        internal static Aes Create(byte[] key)
+        {
+            var aes = Aes.Create();
+
+            aes.Mode = CipherMode.ECB;
+            aes.Padding = PaddingMode.None;
+            aes.KeySize = key.Length * 8;
+            aes.Key = key;
+
+            return aes;
+        }
+
+        public static ICryptoTransform CreateEncryptor(byte[] key)
+        {
+            return CreateEncryptor(key, null);
+        }
+
+        public static ICryptoTransform CreateEncryptor(byte[] key, byte[] iv)
+        {
+            if (key == null)
+                throw new ArgumentNullException(nameof(key));
+
+            if (key.Length != 128 >> 3 && key.Length != 192 >> 3 && key.Length != 256 >> 3)
+                throw new ArgumentException("key length must be 128, 192 or 256 bits");
+
+            if (iv != null && iv.Length != 8)
+                throw new ArgumentException("iv length must be 64 bits");
+
+            return new AesKwEncryptor(key, iv ?? DefaultIv);
+        }
+
+        public static ICryptoTransform CreateDecryptor(byte[] key)
+        {
+            return CreateDecryptor(key, null);
+        }
+
+        public static ICryptoTransform CreateDecryptor(byte[] key, byte[] iv)
+        {
+            if (key == null)
+                throw new ArgumentNullException(nameof(key));
+
+            if (key.Length != 128 >> 3 && key.Length != 192 >> 3 && key.Length != 256 >> 3)
+                throw new ArgumentException("key length must be 128, 192 or 256 bits");
+
+            if (iv != null && iv.Length != 8)
+                throw new ArgumentException("iv length must be 64 bits");
+
+            return new AesKwDecryptor(key, iv ?? DefaultIv);
+        }
+
+        private static byte[] DefaultIv
+        {
+            get { return (byte[])_defaultIv.Clone(); }
+        }
+
+        private static byte[] GetBytes(UInt64 i)
+        {
+            byte[] temp = BitConverter.GetBytes(i);
+
+            if (BitConverter.IsLittleEndian)
+            {
+                Array.Reverse(temp);
+            }
+
+            return temp;
+        }
+
+        private class AesKwEncryptor : ICryptoTransform
+        {
+            private Aes _aes;
+            private byte[] _iv;
+
+            internal AesKwEncryptor(byte[] keyBytes, byte[] iv)
+            {
+                // Create the AES provider
+                _aes = AesKw.Create(keyBytes);
+
+                // Set the AES IV to Zeroes
+                var aesIv = new byte[_aes.BlockSize >> 3];
+
+                aesIv.Zero();
```

Comment:
This shouldn't be necessary (initialized to zeroes during allocation above).

Reply:
This code is ported from directly track 1 and I'm leaving all the code in this class as is, since it's already been security reviewed for it's crypto implementation.